### PR TITLE
uf-10998: Scheduled job for triggering dangling batches.

### PR DESCRIPTION
**Pull request by mha109**
## Types of changes

- [ ] Bug fix
- [x] New feature
- [ ] Removed feature
- [ ] Code style update (formatting etc.)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes

## Does this PR introduce a breaking change?
- [ ] Yes (I have stepped the version number accordingly)
- [x] No
      
## Checklist:

- [x] My code follows the code style of this project.
- [x] I have updated the documentation accordingly (if applicable).
- [x] I have added/updated tests to cover my changes (if applicable).

### CODE REVIEW

File: src/integration-test/java/apptest/SnailMailIT.java [hunk 1]
```diff
@@ -6,18 +6,17 @@
...
 import org.springframework.test.context.jdbc.Sql;
 import org.testcontainers.containers.DockerComposeContainer;
 import org.testcontainers.junit.jupiter.Container;
 import org.testcontainers.junit.jupiter.Testcontainers;
-
 import se.sundsvall.dept44.test.AbstractAppTest;
 import se.sundsvall.dept44.test.annotation.wiremock.WireMockAppTestSuite;
 import se.sundsvall.snailmail.Application;
 import se.sundsvall.snailmail.integration.db.BatchRepository;
+import se.sundsvall.snailmail.service.BatchScheduler;
```
Comment (josc04) [inline src/integration-test/java/apptest/SnailMailIT.java:19]: Is this import used?

File: src/main/java/se/sundsvall/snailmail/Application.java [hunk 2]
```diff
@@ -3,13 +3,14 @@
...
 import org.springframework.cloud.openfeign.EnableFeignClients;
-
+import org.springframework.scheduling.annotation.EnableScheduling;
 import se.sundsvall.dept44.ServiceApplication;
 import se.sundsvall.dept44.util.jacoco.ExcludeFromJacocoGeneratedCoverageReport;
 
 @ServiceApplication
 @EnableFeignClients
 @ExcludeFromJacocoGeneratedCoverageReport
+@EnableScheduling
```
Comment (josc04) [inline src/main/java/se/sundsvall/snailmail/Application.java:13]: Add `@EnableSchedulerLock(defaultLockAtMostFor = "PT5M")` to enable shedlock?

File: src/main/java/se/sundsvall/snailmail/service/BatchScheduler.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+		this.batchRepository = batchRepository;
+		this.snailMailService = snailMailService;
+	}
+
+	/**
+	 * Check for batches that has not been triggered, for whatever reason.
+	 * It is outdated if the batch was created before the configured duration.
+	 */
+	@Scheduled(fixedRateString = "${batch.dangling.check-interval}")
+	@SchedulerLock(name = "${batch.dangling.name}", lockAtMostFor = "${batch.dangling.lock-at-most-for}")
```
Comment (josc04) [inline src/main/java/se/sundsvall/snailmail/service/BatchScheduler.java:34]: I think you need a scheduler configuration like
```
@Configuration
class ShedLockConfiguration {

	@Bean
	LockProvider lockProvider(DataSource dataSource) {
		return new JdbcTemplateLockProvider(
			JdbcTemplateLockProvider.Configuration.builder()
				.usingDbTime()
				.withJdbcTemplate(new JdbcTemplate(dataSource))
				.build());
	}
}
```
 to make shedlock work.

To verify functionality, create a shedlock test (see se.sundsvall.billingpreprocessor.service.scheduler.InvoiceFileSchedulerCreateFilesShedlockTest in BPP for example).

File: src/main/java/se/sundsvall/snailmail/service/BatchScheduler.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+@Component
+public class BatchScheduler {
+
+	private static final Logger LOGGER = LoggerFactory.getLogger(BatchScheduler.class);
+
+	private final Duration outdatedAfter;
+	private final BatchRepository batchRepository;
+	private final SnailMailService snailMailService;
+
+	public BatchScheduler(@Value("${batch.dangling.outdated-after}") final Duration outdatedAfter, BatchRepository batchRepository, SnailMailService snailMailService) {
```
Comment (jklingst) [inline src/main/java/se/sundsvall/snailmail/service/BatchScheduler.java:23]: Use `@ConfigurationProperties` and `@Validated` in conjunction with appropriate `jakarta.validation.*` constraints for a basic sanity/null check on the `batch.dangling.*` properties?

File: src/main/java/se/sundsvall/snailmail/service/BatchScheduler.java (hunk 3 above)
Comment (mha109) [reply src/main/java/se/sundsvall/snailmail/service/BatchScheduler.java:34]: Totally forgot. I'll fix

File: src/integration-test/java/apptest/SnailMailIT.java (hunk 1 above)
Comment (mha109) [reply src/integration-test/java/apptest/SnailMailIT.java:19]: nope

**Review by josc04 — APPROVED** [review]

**Pull request merged**